com.unity.render-pipelines.core/Runtime/Volume/VolumeParameter.cs

```diff
@@ -1641,14 +1642,239 @@ internal override void Interp(VolumeParameter from, VolumeParameter to, float t)
+                float endTime = Mathf.Max(lhsCurve.keys[lhsCurve.length - 1].time, rhsCurve.keys[rhsCurve.length - 1].time);
+
+                // we don't know how many keys the resulting curve will have (because we will compact keys that are at the exact
+                // same time), but in most cases we will need the worst case number of keys. So allocate the worst case, and resize
+                // the array at the end if we need to compact.
+                int maxNumKeys = lhsCurve.length + rhsCurve.length;
+                int currNumKeys = 0;
+                Keyframe[] dstKeys = new Keyframe[maxNumKeys];
+
+                int lhsKeyCurr = 0;
+                int rhsKeyCurr = 0;
+
+                while (lhsKeyCurr < lhsCurve.keys.Length && rhsKeyCurr < rhsCurve.keys.Length)
+                {
+                    // the index is considered invalid once it goes off the end of the array
+                    bool lhsValid = lhsKeyCurr < lhsCurve.keys.Length;
+                    bool rhsValid = rhsKeyCurr < rhsCurve.keys.Length;
+
+                    // it's actually impossible for lhsKey/rhsKey to be uninitialized, but have to
+                    // add initialize here to prevent compiler erros
+                    Keyframe lhsKey = new Keyframe();
+                    Keyframe rhsKey = new Keyframe();
+                    if (lhsValid && rhsValid)
+                    {
+                        lhsKey = lhsCurve.keys[lhsKeyCurr];
+                        rhsKey = rhsCurve.keys[rhsKeyCurr];
+
+                        if (lhsKey.time == rhsKey.time)
+                        {
+                            lhsKeyCurr++;
+                            rhsKeyCurr++;
+                        }
+                        else if (lhsKey.time < rhsKey.time)
+                        {
+                            // in this case:
+                            //     rhsKey[curr-1].time <= lhsKey.time <= rhsKey[curr].time
+                            // so interpolate rhsKey at the lhsKey.time.
+                            rhsKey = EvalKeyAtTime(rhsCurve.keys, rhsKeyCurr - 1, rhsKeyCurr, startTime, endTime, lhsKey.time);
+                            lhsKeyCurr++;
+                        }
+                        else if (lhsKey.time > rhsKey.time)
+                        {
+                            // this is the reverse of the lhs key case
+                            //     lhsKey[curr-1].time <= rhsKey.time <= lhsKey[curr].time
+                            // so interpolate lhsKey at the rhsKey.time.
+                            lhsKey = EvalKeyAtTime(lhsCurve.keys, lhsKeyCurr - 1, lhsKeyCurr, startTime, endTime, rhsKey.time);
+                            rhsKeyCurr++;
+                        }
+                        else
+                        {
+                            // should never happen, checking here to be safe
+                            Assert.IsTrue(false);
+                        }
```

Comment:
This effectively cannot happen
if(x == y) ... //as time is float, equality of float can be tricky. Try perhaps Mathf.Approximately
else if (x < y) ...
else if (x > y) ... //can remove the condition
else ... //can be removed

Reply:
Generally, I like to keep the else in even if it can't happen with an assert, so if someone changes the code later it's more explicitly clear. But if it's easier, I'll just remove the else and put in a comment. The first (x==y) is by design as even if x and y are very close except for a small epsilon, it's still preferable to add two separate points. I thought about trying to merge points based on an epsilon, but it creates a bunch of corner cases where the interpolation might pop in unexpected ways.

Reply:
I personally found the else confusing and there is an added unneeded test even if the cost is really small.
This is code that we will maintain so please do the change. It will be simpler for us to go back onto it.

For the float equality, do as you prefer for this. I don't know exactly what is your issue here but I believe you have think about it and think it is better this way.

Reply:
Sounds good. I'll fix it. My bottleneck is that at the moment, the HDRP sample scene isn't working in my version of the engine, so I'm trying to figure out how to get a sample scene up and running. Then I'll submit the changes.